threedi_schema/migrations/versions/0300_upgrade_db_settings.py

```diff
@@ -0,0 +1,296 @@
+    for src_name, dst_name in table_sets:
+        op.rename_table(src_name, dst_name)
+
+
+def rename_columns(table_name: str, columns: List[Tuple[str, str]]):
+    # no checks for existence are done, this will fail if table or any source column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for src_name, dst_name in columns:
+            batch_op.alter_column(src_name, new_column_name=dst_name)
+
+
+def make_all_columns_nullable(table_name, id_name: str = 'id'):
+    # no checks for existence are done, this will fail if table doesn't exist
+    connection = op.get_bind()
+    table = sa.Table(table_name, sa.MetaData(), autoload_with=connection)
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in table.columns:
+            if column.name == id_name:
+                continue
+            batch_op.alter_column(column_name=column.name, nullable=True)
+
+
+def create_new_tables(table_names: List[str]):
+    # no checks for existence are done, this will fail if any table already exists
+    for table_name in table_names:
+        op.create_table(table_name, sa.Column("id", sa.Integer(), primary_key=True))
+
+
+def add_columns_to_tables(table_columns: List[Tuple[str, Column]]):
+    # no checks for existence are done, this will fail if any column already exists
+    for dst_table, col in table_columns:
+        with op.batch_alter_table(dst_table) as batch_op:
+            batch_op.add_column(col)
+
+
+def move_multiple_values_to_empty_table(src_table: str, dst_table: str, columns: List[str]):
+    # move values from one table to another
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    dst_cols = ', '.join(dst for _, dst in columns)
+    src_cols = ', '.join(src for src, _ in columns)
+    op.execute(sa.text(f'INSERT INTO {dst_table} ({dst_cols}) SELECT {src_cols} FROM {src_table}'))
+    remove_columns_from_table(src_table, [src for src, _ in columns])
+
+
+def move_values_to_table(src_table: str, src_col: str, dst_table: str, dst_col: str):
+    op.execute(f'UPDATE {dst_table} SET {dst_col} = (SELECT {src_col} FROM {src_table} LIMIT 1)')
+    remove_columns_from_table(src_table, [src_col])
+
+
+def remove_columns_from_table(table_name: str, columns: List[str]):
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_settings_id():
+    for settings_col, settings_id, settings_table in GLOBAL_SETTINGS_ID_TO_BOOL:
+        if settings_col != '':
+            # set boolean 'use_*' in model_settings if a relationship exists
+            op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE {settings_id} IS NOT NULL;")
+        # remove all settings rows, exact for the one matching
```

Comment:
Why not dropping the tables alltogether? Simpler for sure.

Reply:
I don't understand how dropping the table would be a solution because we want to keep the column and one row.

Reply:
Right, I get it.